controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -306,36 +303,32 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	kcp.Status.UpdatedReplicas = int32(len(currentMachines))
 
 	replicas := int32(len(ownedMachines))
+
+	// set basic data that does not require interacting with the workload cluster
 	kcp.Status.Replicas = replicas
-	readyMachines := int32(0)
+	kcp.Status.ReadyReplicas = 0
+	kcp.Status.UnavailableReplicas = 0
```

Comment:
Will this wipe out the "last known" status in case of a transient connection issue?

Reply:
this should be not be 0 but should be `replicas`. However, yes, your'e right this will wipe out last known status in case of transient issues as that was the previous behavior as well.